libs/mbedtls/mbedtls_stubs.c

```diff
@@ -192,6 +192,31 @@ value caml_string_of_asn1_buf(mbedtls_asn1_buf* dat) {
 	CAMLreturn(s);
 }
 
+CAMLprim value hx_cert_get_alt_names(value chain) {
+	CAMLparam1(chain);
+	CAMLlocal1(obj);
+	mbedtls_x509_crt* cert = X509Crt_val(chain);
+	if (cert->ext_types & MBEDTLS_X509_EXT_SUBJECT_ALT_NAME == 0 || &cert->subject_alt_names == NULL) {
+		obj = Atom(0);
```

Comment:
@Aurel300 fyi, I found another potential trap here:

> Atom(t) returns an “atom” (zero-sized block) with tag t. Zero-sized blocks are preallocated outside of the heap. It is incorrect to try and allocate a zero-sized block using the functions below. For instance, Atom(0) represents the empty array.

This means that whenever we want to return an `array` from C externs, we have to deal with the `length == 0` special case.